pkg/cvo/configuration/configuration.go

```diff
@@ -0,0 +1,157 @@
+	// queue tracks checking for the CVO configuration.
+	//
+	// The type any is used to comply with the worker method of the cvo.Operator struct.
+	queue workqueue.TypedRateLimitingInterface[any]
+
+	client  cvoclientv1alpha1.ClusterVersionOperatorInterface
+	lister  operatorlistersv1alpha1.ClusterVersionOperatorLister
+	factory operatorexternalversions.SharedInformerFactory
+
+	started bool
+
+	desiredLogLevel        operatorv1.LogLevel
+	lastObservedGeneration int64
+}
+
+func (config *ClusterVersionOperatorConfiguration) Queue() workqueue.TypedRateLimitingInterface[any] {
+	return config.queue
+}
+
+// clusterVersionOperatorEventHandler queues an update for the cluster version operator on any change to the given object.
+// Callers should use this with an informer.
+func (config *ClusterVersionOperatorConfiguration) clusterVersionOperatorEventHandler() cache.ResourceEventHandler {
+	return cache.ResourceEventHandlerFuncs{
+		AddFunc: func(_ interface{}) {
+			config.queue.Add(config.queueKey)
+			klog.V(internal.Debug).Infof("ClusterVersionOperator resource was added; queuing a configuration sync")
+		},
+		UpdateFunc: func(_, _ interface{}) {
+			config.queue.Add(config.queueKey)
+			klog.V(internal.Debug).Infof("ClusterVersionOperator resource was modified or resync period has passed; queuing a configuration sync")
+		},
+		DeleteFunc: func(_ interface{}) {
+			klog.V(internal.Debug).Infof("ClusterVersionOperator resource was deleted; configuration sync is not queued")
+		},
+	}
+}
+
+// NewClusterVersionOperatorConfiguration returns ClusterVersionOperatorConfiguration, which might be used
+// to synchronize with the ClusterVersionOperator resource.
+func NewClusterVersionOperatorConfiguration(client operatorclientset.Interface, factory operatorexternalversions.SharedInformerFactory) *ClusterVersionOperatorConfiguration {
+	return &ClusterVersionOperatorConfiguration{
+		queueKey: fmt.Sprintf("ClusterVersionOperator/%s", ClusterVersionOperatorConfigurationName),
+		queue: workqueue.NewTypedRateLimitingQueueWithConfig[any](
+			workqueue.DefaultTypedControllerRateLimiter[any](),
+			workqueue.TypedRateLimitingQueueConfig[any]{Name: "configuration"}),
+		client:  client.OperatorV1alpha1().ClusterVersionOperators(),
+		factory: factory,
+	}
+}
+
+// Start initializes and starts the configuration's informers. Must be run before Sync is called.
+// Blocks until informers caches are synchronized or the context is cancelled.
+func (config *ClusterVersionOperatorConfiguration) Start(ctx context.Context) error {
+	informer := config.factory.Operator().V1alpha1().ClusterVersionOperators()
+	if _, err := informer.Informer().AddEventHandler(config.clusterVersionOperatorEventHandler()); err != nil {
+		return err
+	}
+	config.lister = informer.Lister()
+
+	config.factory.Start(ctx.Done())
+	synced := config.factory.WaitForCacheSync(ctx.Done())
+	for _, ok := range synced {
+		if !ok {
+			return fmt.Errorf("caches failed to sync: %w", ctx.Err())
+		}
+	}
```

Comment:
Note: Blocks the controller till the `ClusterVersionOperator` CRD is present in the cluster. When transitioning from Default to DevPreview, this will create temporally errors till the payload sync loop finishes. We may use a discovery client to check rather than blocking and logging errors.

Reply:
Errors happen in the intermediate state of the CVO to my knowledge.

- CVO running Default
- CVO notices changes in the `FeatureGate`
- CVO restarts
- CVO starts applying new payload (starts omitting mentioned errors)
- CVO restarts as a new CVO deployment is applied
- CVO is happy (no errors)